lib/modules/cardiovascular_disease.rb

```diff
@@ -366,16 +407,72 @@ def self.perform_encounter(entity, time)
             patient.condition(diagnosis, time, :condition, :condition)
           end
         end
+
+        if entity[:careplan] && entity[:careplan][:cardiovascular_disease]
+          if !entity.record_synthea.careplan_active?(:cardiovascular_disease)
+            entity.record_synthea.careplan_start(:cardiovascular_disease, entity[:careplan][:cardiovascular_disease]['activities'], time, entity[:careplan][:cardiovascular_disease]['reasons'])
+          else 
+            entity.record_synthea.update_careplan_reasons(:cardiovascular_disease, entity[:careplan][:cardiovascular_disease]['reasons'],time)
+          end
+        elsif entity.record_synthea.careplan_active?(:cardiovascular_disease)
+          entity.record_synthea.careplan_stop(:cardiovascular_disease,time)
+        end
+
+        if entity[:medications]
+          [:clopidogrel, :simvastatin, :amlodipine, :nitroglycerin, :atorvasatatin, :captopril, :warfarin, 
+            :verapamil, :digoxin, :epinephrine, :amiodarone, :atropine].each do |med|
+            if entity[:medications][med]
+              # Add a prescription to the record if it hasn't been recorded yet
+              if !entity.record_synthea.medication_active?(med)
+                entity.record_synthea.medication_start(med, time, entity[:medications][med][1])
```

Comment:
I think it would be easier to read (and author) if you used a hash to store time and reason, rather than array (that you have to remember which index is which).  This (obviously) goes for anywhere you're using `entity[:medications][med][1]` so I won't repeat myself in the other cases.